File: _includes/code/python/config-refs.datatypes.blob.py
import weaviate

# START ConfigureDataType
from weaviate.classes.config import Property, DataType

# END ConfigureDataType

client = weaviate.connect_to_local()

# Delete collection if it exists
client.collections.delete(name="Poster")

# START ConfigureDataType
# Create collection
my_collection = client.collections.create(
    name="Poster",
    properties=[
        Property(name="title", data_type=DataType.TEXT),
        Property(name="image", data_type=DataType.BLOB),
    ],
    # Other properties are omitted for brevity
)
# END ConfigureDataType

blob_string = "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8z8BQDwAEhQGAhKmMIQAAAABJRU5ErkJggg=="

# START AddObject
# Create an object
example_object = {
    "title": "The Matrix",
    "image": blob_string
}

obj_uuid = my_collection.data.insert(example_object)
# END AddObject

returned_obj = my_collection.query.fetch_object_by_id(obj_uuid, return_properties=["title", "image"])

assert obj_uuid is not None
assert returned_obj.properties == example_object

client.close()

File: _includes/code/typescript/config-refs.datatypes.blob.ts
import assert from 'assert';
import weaviate, { WeaviateClient } from 'weaviate-client';

// START ConfigureDataType
import { dataType } from 'weaviate-client';

// END ConfigureDataType

const client: WeaviateClient = await weaviate.connectToLocal();

await client.collections.delete('Poster');

// START ConfigureDataType
// Create collection
const myCollection = await client.collections.create({
  name: 'Poster',
  properties: [
    {
      name: 'title',
      dataType: dataType.TEXT,
    },
    {
      name: 'image',
      dataType: dataType.BLOB,
    },
  ],
  // Other properties are omitted for brevity
});
// END ConfigureDataType

const blob_string = 'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8z8BQDwAEhQGAhKmMIQAAAABJRU5ErkJggg==';

// START AddObject
const exampleObject = {
  title: 'The Matrix',
  image: blob_string
};

const obj_uuid = await myCollection.data.insert(exampleObject);
// END AddObject

const returnedObject = await myCollection.query.fetchObjectById(obj_uuid, {
  returnProperties: ['title', 'image']  // Need to specify `image` to get the blob
});

assert.notEqual(obj_uuid, null);
assert.deepEqual(returnedObject?.properties, exampleObject);

client.close();
